File: jimmer-ddd-java-sample/admin/src/main/java/org/morecup/jimmerddd/java/sample/admin/domain/OrderRepositoryImpl.java
package org.morecup.jimmerddd.java.sample.admin.domain;

import org.morecup.jimmerddd.core.preanalysis.MethodInfo;
import org.morecup.jimmerddd.java.sample.domain.base.DomainException;
import org.morecup.jimmerddd.java.sample.domain.order.Order;
import org.morecup.jimmerddd.java.sample.domain.order.OrderRepository;
import org.morecup.jimmerddd.java.spring.preanalysis.JAggregateSqlClient;
import org.springframework.stereotype.Component;

@Component
public class OrderRepositoryImpl implements OrderRepository {

    private final JAggregateSqlClient aggregateSqlClient;

    public OrderRepositoryImpl(JAggregateSqlClient aggregateSqlClient) {
        this.aggregateSqlClient = aggregateSqlClient;
    }

    @Override
    public Order saveOrder(Order order) {
        return aggregateSqlClient.saveAggregate(order).getModifiedEntity();
    }

    @Override
    public Order findByIdOrErr(long id, MethodInfo function) {
        Order result = aggregateSqlClient.findById(Order.class,function,id);
        if (result == null) {
            throw new DomainException("找不到，id: " + id);
        }
        return result;
    }
}

File: jimmer-ddd-kotlin-sample/admin/src/main/kotlin/org/morecup/jimmerddd/kotlin/sample/admin/domain/OrderRepositoryImpl.kt
package org.morecup.jimmerddd.kotlin.sample.admin.domain

import org.babyfish.jimmer.spring.repo.support.AbstractKotlinRepository
import org.babyfish.jimmer.sql.ast.mutation.AssociatedSaveMode
import org.babyfish.jimmer.sql.ast.mutation.SaveMode
import org.babyfish.jimmer.sql.kt.KSqlClient
import org.morecup.jimmerddd.kotlin.sample.domain.base.DomainException
import org.morecup.jimmerddd.kotlin.sample.domain.order.Order
import org.morecup.jimmerddd.kotlin.sample.domain.order.OrderRepository
import org.morecup.jimmerddd.kotlin.spring.preanalysis.findById
import org.morecup.jimmerddd.kotlin.spring.preanalysis.saveAggregate
import org.springframework.stereotype.Component
import kotlin.reflect.KFunction

@Component
class OrderRepositoryImpl(
    sql: KSqlClient
) : AbstractKotlinRepository<Order, Long>(sql), OrderRepository {

    override fun saveOrder(order: Order): Order {
        return saveAggregate(order).modifiedEntity
    }

    override fun findByIdOrErr(id: Long,function: KFunction<*>?): Order {
        return findById(id,function) ?: throw DomainException("找不到，id: $id")
    }

}